runtime/src/qio/qio.c

```diff
@@ -2553,8 +2553,8 @@ qioerr _qio_buffered_behind(qio_channel_t* ch, int flushall)
           break;
         // no default to get warnings when new methods are added
       }
-      if( err ) goto error;
       qbuffer_iter_advance(&ch->buf, &write_start, num_written);
+      if( err ) goto error;
```

Comment:
Is this code intended to solve an off by one error with the qbuffer_iter?  We don't want the iter to be pointing at where the error occurred, we want it to point beyond where we landed?